src/fusion_power_plant.h

```diff
@@ -59,27 +56,170 @@ class FusionPowerPlant : public cyclus::Facility  {
+    "uilabel": "Recipe for blanket feed material" \
+  }
+  std::string blanket_inrecipe;
+
+  #pragma cyclus var { \
+    "doc": "Blanket waste commodity designation", \
+    "tooltip": "Blanket waste commodity", \
+    "uilabel": "Blanket waste commodity" \
+  }
+  std::string blanket_outcommod;
+
+  #pragma cyclus var { \
+    "default": 1000.0, \
+    "doc": "Initial mass of full blanket material", \
+    "tooltip": "Only blanket material mass, not structural mass", \
+    "units": "kg", \
+    "uitype": "range", \
+    "range": [0, 10000], \
+    "uilabel": "Initial Mass of Blanket" \
+  }
+  double blanket_size;
+
+  #pragma cyclus var { \
+    "default": 0.05, \
+    "doc": "Percent of blanket that gets recycled every blanket turnover period", \
+    "tooltip": "Defaults to 0.05 (5%), must be between 0 and 15%", \
+    "units": "dimensionless", \
+    "uitype": "range", \
+    "range": [0, 0.15], \
+    "uilabel": "Blanket Turnover Rate" \
+  }
+  double blanket_turnover_rate;
+
+  //Resource Buffers and Trackers:
   cyclus::toolkit::ResBuf<cyclus::Material> tritium_storage;
   cyclus::toolkit::ResBuf<cyclus::Material> tritium_excess;
   cyclus::toolkit::ResBuf<cyclus::Material> helium_excess;
   cyclus::toolkit::ResBuf<cyclus::Material> blanket_feed;
   cyclus::toolkit::ResBuf<cyclus::Material> blanket_waste;
 
-  //PyneIDs
-  const int tritium_id = 10030000;
+  cyclus::toolkit::MatlBuyPolicy fuel_startup_policy;
+  cyclus::toolkit::MatlBuyPolicy fuel_refill_policy;
+  cyclus::toolkit::MatlBuyPolicy blanket_fill_policy;
 
-  //Compositions:
-  const cyclus::CompMap T = {{tritium_id, 1}};
-  const cyclus::Composition::Ptr tritium_comp = cyclus::Composition::CreateFromAtom(T);
+  cyclus::toolkit::MatlSellPolicy tritium_sell_policy;
+  cyclus::toolkit::MatlSellPolicy helium_sell_policy;
+  cyclus::toolkit::MatlSellPolicy blanket_waste_sell_policy;
 
-  //Materials:
-  cyclus::Material::Ptr sequestered_tritium = cyclus::Material::CreateUntracked(0.0, tritium_comp);
+  cyclus::toolkit::TotalInvTracker fuel_tracker;
+  cyclus::toolkit::TotalInvTracker blanket_tracker;
+
+  //Functions:
+  void CycleBlanket();
+  bool CheckOpeartingConditions();
+  bool CheckInitialStartupCondition();
+  void SequesterTritium();
+  void OperateReactor();
+  void CycleBlanket();
+  void DecayInventories();
+  void ExtractHelium();
+  void MoveExcessTritiumToSellBuffer();
+
+  private:
+    bool initial_startup_condition_met = false;
+    double fuel_usage_atoms;
+    double fuel_usage_mass;
+    const double seconds_per_year = 2629846*12;
+    const double MW_to_GW = 1000.0;
+    double avagadros_number = 6.022e23;
+    double fuel_limit = 1000.0;
+    double blanket_limit = 100000.0;
+
+
+    //NucIDs for Pyne
+    const int tritium_id = 10030000;
+    const int He3_id = 20030000;
+    const int He4_id = 20040000;
+    const int Li6_id = 30060000;
+    const int Li7_id = 30070000;
+
+    //masses in kg/atom
+    const double amu_to_kg = 1.66054e-27;
```

Comment:
Another quantity we shouldn't really need here

Reply:
Because we should get it from pyne?

Reply:
Or avoid using it at all by using PyNE's ability to translate from mass to atoms (maybe?)

Reply:
I was unaware of this capability (I don't typically use pyne). I'll leave this in here for now so things can get up and running, but make a comment and a to-do item to try to switch to using some pyne features to do this type of operation instead, if that works for you?